pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,275 @@
+
+// NewProvider instantiates Provider
+func NewProvider(dbPath string) (*Provider, error) {
+	if dbPath == "" {
+		return nil, errors.New(blankDBPathErrMsg)
+	}
+
+	// Example DB Path root:my-secret-pw@tcp(127.0.0.1:3306)/
+	db, err := sql.Open("mysql", dbPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open connection: %w", err)
+	}
+
+	p := &Provider{
+		dbPath: dbPath,
+		db:     db,
+		dbs:    map[string]*sqlDBStore{}}
+
+	return p, nil
+}
+
+// OpenStore opens and returns new db for given name space.
+func (p *Provider) OpenStore(name string) (storage.Store, error) {
+	p.Lock()
+	defer p.Unlock()
+
+	tx, err := p.db.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	// creating the database
+	_, err = tx.Exec(createDBQuery + name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create db %s: %w", name, err)
+	}
+
+	// Use is used to select the created database without this DDL operations are not permitted
+	_, err = tx.Exec(useDBQuery + name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to use db %s: %w", name, err)
+	}
+
+	tableName := tablePrefix + name
+	// TODO: Issue-1940 Store the hashed key to control the width of the key varchar column
+	createTableStmt := "CREATE Table IF NOT EXISTS " + tableName +
+		"(`key` varchar(255) NOT NULL ,`value` BLOB, PRIMARY KEY (`key`));"
+
+	// creating key-value table inside the database
+	_, err = tx.Exec(createTableStmt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create table %s: %w", tableName, err)
+	}
+
+	store := &sqlDBStore{
+		db:        p.db,
+		tableName: tableName}
+
+	p.dbs[name] = store
+
+	return store, nil
+}
+
+// Close closes the provider.
+func (p *Provider) Close() error {
+	p.Lock()
+	defer p.Unlock()
+
+	for _, store := range p.dbs {
+		err := store.db.Close()
+		if err != nil {
+			return fmt.Errorf(failToCloseProviderErrMsg+": %w", err)
+		}
+	}
+
+	if err := p.db.Close(); err != nil {
+		return err
+	}
+
+	p.dbs = make(map[string]*sqlDBStore)
+
+	return nil
+}
+
+// CloseStore closes a previously opened store
+func (p *Provider) CloseStore(name string) error {
+	p.Lock()
+	defer p.Unlock()
+
+	store, exists := p.dbs[name]
+	if !exists {
+		return nil
+	}
+
+	delete(p.dbs, name)
+
+	return store.db.Close()
+}
+
+// Put stores the key and the record
+func (s *sqlDBStore) Put(k string, v []byte) error {
+	if k == "" {
+		return errKeyRequired
+	}
+
+	//nolint: gosec create upsert query to insert the record, checking whether the key is already mapped to a value in the store.
+	createStmt := "INSERT INTO " + s.tableName + " VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?"
+	// executing the prepared insert statement
+	_, err := s.db.Exec(createStmt, k, v, v)
+	if err != nil {
+		return fmt.Errorf("failed to insert key and value record into %s %w ", s.tableName, err)
+	}
+
+	return nil
+}
+
+// Get fetches the record based on key
+func (s *sqlDBStore) Get(k string) ([]byte, error) {
+	if k == "" {
+		return nil, errKeyRequired
```

Comment:
I feel better to use consistent error messages across the implementations. May be by defining it in `storage` pkg.